### PR TITLE
Issue 5696 fixes

### PR DESCRIPTION
**Pull request by VivekVinushanth**
Issue 5696 : https://github.com/wso2/product-is/issues/5696

Method signature changed as required.
tenant domain can be taken from tenantId,

### CODE REVIEW

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java [hunk 1]
```diff
@@ -188,16 +194,21 @@ public static void setRealmService(RealmService realmService) {
...
+     *
+     */
+    @Deprecated
+    public static void initializeRegistry(int tenantId, String tenantDomain) throws IdentityException {
+        initializeRegistry(tenantId);
     }
 
-
-    public static void initializeRegistry(int tenantId, String tenantDomain) throws IdentityException {
+    public static void initializeRegistry(int tenantId) throws IdentityException {
```
Comment (janakamarasena) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java]: Let's not modify the method signature of an existing public method. Instead, add an overloaded method and call it from within the old method

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java [hunk 2]
```diff
@@ -58,11 +58,14 @@ public static TenantRegistryLoader getTenantRegistryLoader() {
         return tenantRegistryLoader;
     }
 
+    public static void setTenantRegistryLoader(TenantRegistryLoader tenantRegistryLoader) {
```
Comment (ruwanta) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java]: From where this method is called?
It is not safe to set a static variable via public method as anyone can set to any value , thus system can break or can lead to race conditions.

Comment (madurangasiriwardena) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java]: This seems to be an existing method. After the refactoring, it is showing as a new method. At the moment it is called from the service component when it gets a refference to `TenantRegistryLoader` service instance.

Comment (ruwanta) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java]: ack

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java [hunk 3]
```diff
@@ -164,13 +167,16 @@ public static RegistryService getRegistryService() {
         return registryService;
     }
 
+    public static void setRegistryService(RegistryService registryService) {
```
Comment (GANGANI) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java:170]: Add a new line after public methods. Check all the places. Let's add method comments for public methods.

Comment (VivekVinushanth) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java:170]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/3767

**Pull request merged**